### DISCUSSION TITLE
Is there any way to swipe from one point to another point consistently?

### DISCUSSION THREAD

**Discussion by rafi4135** [appium/appium#18705] [Q&A]
I am trying to generate menu tree for an app. I want to swipe vertically from one point to another for many times in the same page from top. But I observed that swipe is not consistent, same swipe method with same parameters makes different swipe actions each time. Sometimes swipe from top uncovers 2 elements, sometime 3 elements. I am using this method:
`mobile_driver.swipe(start_x, start_y, end_x, end_y)`
is there a solution?

**Comment by jlipps** [comment]
Try to use the Actions class instead, it provides fine grained control.

**Reply by rafi4135** [reply]
Will it be consistent?

**Reply by jlipps** [reply]
What Appium sends to the device will be consistent. What the device does with it after that is out of our hands. Slower and more precise actions will tend to be more consistent. But let's say you do a quick flick. Of course you're not guaranteed to scroll the exact same number of pixels each time; that's up to how the device interprets your gesture.
